hack/test-go.sh

```diff
@@ -77,22 +77,45 @@ fi
 
 KUBE_TIMEOUT=${KUBE_TIMEOUT:--timeout 60s}
 
-if [ "${1-}" != "" ]; then
-  if [[ "${1}" == *"/..." ]]; then
+# Declare and set arrays to null string. We need to always keep the first
+# element to be the null string, otherwise the variable is considered unset.
+# When using, slice it to ignore the first element: ${varname[@]:1}.
+declare -a test_flags= test_packages=
```

Comment:
In case anyone's still watching this ... the *real* correct way to do this which
 - correctly passes `('')` as an array of one element, the empty string
 - correctly passes `()` as an empty array
 - doesn't split the array members on spaces

```sh
$ my_function "${array[@]+"${array[@]}"}"
```